python/anvill/binja.py

```diff
@@ -173,8 +173,53 @@ def _convert_bn_llil_type(
         ret = IntegerType(reg_size_bytes, True)
         return ret
 
+def _convert_named_type_reference(
+    bv, tinfo: bn.types.Type, cache) -> Type:
+    """ Convert named type references into a `Type` instance"""
+    if tinfo.type_class != bn.TypeClass.NamedTypeReferenceClass:
+        return
+
+    named_tinfo = tinfo.named_type_reference
+    if (named_tinfo.type_class
+        == bn.NamedTypeReferenceClass.StructNamedTypeClass):
+        # Get the bn struct type and recursively recover the elements
+        ref_type = bv.get_type_by_name(named_tinfo.name);
+        struct_type =  ref_type.structure
+        ret = StructureType()
+        for elem in struct_type.members:
+            ret.add_element_type(_convert_bn_type(bv, elem.type, cache))
+        return ret
+
+    elif (named_tinfo.type_class
+        == bn.NamedTypeReferenceClass.UnionNamedTypeClass):
+        # Get the union type and recover the member elements
+        ref_type = bv.get_type_by_name(named_tinfo.name);
+        struct_type =  ref_type.structure
+        ret = UnionType()
+        for elem in struct_type.union.members:
+            ret.add_element_type(_convert_bn_type(bv, elem.type, cache))
+        return ret
+
+    elif (named_tinfo.type_class
+        == bn.NamedTypeReferenceClass.TypedefNamedTypeClass):
+        ref_type = bv.get_type_by_name(named_tinfo.name);
+        ret = TypedefType()
+        ret.set_underlying_type(_convert_bn_type(bv, ref_type, cache))
+        return ret
+
+    elif (named_tinfo.type_class
+        == bn.NamedTypeReferenceClass.EnumNamedTypeClass):
+        # Set the underlying type int of size width
+        ref_type = bv.get_type_by_name(named_tinfo.name);
+        ret = EnumType()
+        ret.set_underlying_type(IntegerType(ref_type.width, False))
+        return ret
+
+    else:
+        DEBUG("WARNING: Unknown named type {} not handled".format(named_tinfo))
 
-def _convert_bn_type(tinfo: bn.types.Type, cache):
+
+def _convert_bn_type(bv, tinfo: bn.types.Type, cache):
     """Convert an bn `Type` instance into a `Type` instance."""
     if str(tinfo) in cache:
```

Comment:
Can you add a function, `_cache_key`, that we will use in place of `str`, for converting a binary ninja type into a key for the cache, and implement as `str` for now, but so that it's easier to swap out later.